### PR TITLE
Add info about tests that require window focus and `pyautogui` permissions setup on macOS

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

# Description

Add info related with tests that use `pyautogui` and the `skip_local_focus` decorator like the one being done at https://github.com/napari/napari/pull/6699

### CODE REVIEW

File: docs/developers/contributing/testing.md [hunk 1]
```diff
@@ -82,14 +82,16 @@ and are available within
 To run our test suite locally, run `pytest` on the command line.  If, for some reason
 you don't already have the test requirements in your environment, run `python -m pip install -e .[testing]`.
 
-There are some tests that require showing GUI elements, (such
-as testing screenshots). By default, these are only run during continuous integration.
+There are some tests that require showing GUI elements (such
+as testing screenshots) and window focus (such as testing drag and drop behavior). By default, these are only run during continuous integration.
 If you'd like to include them in local tests, set the environment variable "CI":
```
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]:
```suggestion
If you'd like to enable them in local tests, you can set the environment variables: `NAPARI_POPUP_TESTS=1` or `NAPARI_FOCUS_TESTS=1` or set the environment variable `CI=1`:
```

File: docs/developers/contributing/testing.md [hunk 2]
```diff
@@ -82,14 +82,16 @@ and are available within
...
 
 ```sh
 CI=1 pytest
 ```
 
+ Also, if running the tests on macOS, be sure to give the Terminal app `Accessibility` permissions in `System Settings > Privacy & Security > Accessibility` (needed for some tests that rely on `pyautogui`).
```
Comment (psobolewskiPhD) [inline docs/developers/contributing/testing.md]:
```suggestion
Note: setting `CI=1` will also disable certain tests that take too long, etc. on CI.
Also, if running the GUI tests that use `pyautogui` on macOS, be sure to give the Terminal app `Accessibility` permissions in `System Settings > Privacy & Security > Accessibility` so `pyautogui` can control the mouse, keyboard, etc.
```

**Review by psobolewskiPhD — COMMENTED** [review]
Small suggestions, feel free to reword or ignore -- just click resolved!

**Review by psobolewskiPhD — APPROVED** [review]

**Comment by melissawm** [comment]
Looks great, thanks @dalthviz and @psobolewskiPhD !

**Pull request merged**